### PR TITLE
Local result

### PR DESCRIPTION
**Pull request by joshmeek**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
Adds a `LocalResult` type for managing results w/ the local file system

## Why is this PR important?
PIN 16

### CODE REVIEW

File: src/prefect/engine/result/base.py [hunk 1]
```diff
@@ -203,40 +203,40 @@ def deserialize_from_bytes(cls, serialized_value: Union[str, bytes]) -> Any:
 
     def format(self, **kwargs: Any) -> "Result":
         """
-        Takes a set of string format key-value pairs and renders the result.filepath_template to a final filepath string
+        Takes a set of string format key-value pairs and renders the result.location_template to a final location string
```
Comment (cicdw) [inline src/prefect/engine/result/base.py:206]:
```suggestion
        Takes a set of string format key-value pairs and renders the result.location to a final location string
```

File: src/prefect/engine/result/base.py [hunk 2]
```diff
@@ -203,40 +203,40 @@ def deserialize_from_bytes(cls, serialized_value: Union[str, bytes]) -> Any:
...
 
         Args:
-            - **kwargs (Any): string format arguments for result.filepath_template
+            - **kwargs (Any): string format arguments for result.location_template
```
Comment (cicdw) [inline src/prefect/engine/result/base.py:209]:
```suggestion
            - **kwargs (Any): string format arguments for result.location
```

File: tests/engine/results/test_results.py [hunk 3]
```diff
@@ -120,3 +129,65 @@ def test_exists_for_json_objs(self, value):
         result = PrefectResult()
         assert result.exists(json.dumps(value)) is True
         assert result.exists(value) is False
+
+
+class TestLocalResult:
+    @pytest.fixture(scope="class")
+    def tmp_dir(self):
```
Comment (cicdw) [inline tests/engine/results/test_results.py:136]: No need to change this but FYI `pytest` offers a builtin fixture for this pattern `tmpdir` (no underscore) that you can use.

Comment (joshmeek) [reply tests/engine/results/test_results.py:136]: TIL

**Pull request merged**